app/code/community/Bolt/Boltpay/Helper/Api.php

```diff
@@ -273,7 +273,19 @@ public function createOrder($reference, $session_quote_id = null)
 
         // a call to internal Magento service for order creation
         $service = Mage::getModel('sales/service_quote', $quote);
-        $service->submitAll();
+
+        try {
+            $service->submitAll();
+        } catch (Exception $e) {
+            $errorMessage = 'Error submitting order';
+            $metaData = array(
+                'transaction'   => json_encode((array)$transaction),
+                'quote_address' => var_export($quote->getShippingAddress()->debug(), true)
+            );
+            Mage::helper('boltpay/bugsnag')->notifyException(new Exception($errorMessage), $metaData);
```

Comment:
(non-blocking) It's ok for now but I think this would cause multiple error reports (one from here, and the another from upstream). I think what we can do in the long term is defining custom error type so we can propagate all info to upper level and we bugsnag at the top level.